# -*- mode: Snakemake -*-

rule rgi_all:
	input:
		rules.run_rgi.output

rule run_rgi:
	input:
		faa = str(ANNOTATION_FP/'genes'/'prodigal'/'{sample}_genes_prot.fa'),
		rg_db_read = str(ANNOTATION_FP/'card'/'db_in.txt')
	output:
		out = str(ANNOTATION_FP/'card'/'{sample}_rgi'),
		json = str(ANNOTATION_FP/'card'/'{sample}_rgi.json')
	shell:
		"""
		rgi main --input_sequence {input.faa} --output_file {output.out} --input_type protein --low_quality --clean
		"""

rule read_rgi:
	output:
		temp(str(ANNOTATION_FP/'card'/'db_in.txt'))
	params:
		card_json = str(Cfg['sbx_rgi']['card_db_path']/'{genome}.fasta')
	shell:
		"""
		rgi load -i {params.card_json} > {output}
		"""
